Tests/DatadogTests/DatadogObjc/DDNSURLSessionDelegateTests.swift

```diff
@@ -0,0 +1,237 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import XCTest
+@testable import Datadog
+@testable import DatadogObjc
+
+class DDNSURLSessionDelegateTests: XCTestCase {
```

Comment:
Those tests are copy & pasted [from `DDURLSessionDelegate` tests](https://github.com/DataDog/dd-sdk-ios/blob/master/Tests/DatadogTests/Datadog/URLSessionAutoInstrumentation/DDURLSessionDelegateTests.swift) and do not cover anything new than what's covered in swift delegate tests. We should write tests for the new logic we added.

The new behaviours added in `DDNSURLSessionDelegate` are:
* 1/ objc delegate must forward all calls to the wrapped swift delegate,
* 2/ objc delegate's type must appear as the wrapped swift delegate type at runtime.

Additionally, as objc delegate is `open class` and can be subclassed, we should assert that:
* 3/ when subclassing objc delegate it doesn't change the 1st and 2nd behaviour.